lib/vehicle_factory.rb

```diff
@@ -0,0 +1,105 @@
+class VehicleFactory
+  attr_reader :vehicles
+
+  def initialize
+    @vehicles = []
+  end
+
+  def create_vehicles(data_service)
+    data_service.each do |vehicle_info|
+      formatted_vehicle_info = {
+        vin: change_to_vin_key(vehicle_info),
+        year: vehicle_info[:model_year],
+        make: vehicle_info[:make],
+        model: change_to_model_key(vehicle_info),
+        engine: change_to_engine_key(vehicle_info),
+        county: vehicle_info[:county]
+        }
+        @vehicles << Vehicle.new(formatted_vehicle_info)
+    end
+  end
+
+  def change_to_engine_key(vehicle_info)
+    if vehicle_info.has_key?(:electric_vehicle_type)
+      :ev
+    elsif vehicle_info[:fuel_type] == "GAS"
+      :gas
+    else
+      'Unavailable.'
+    end
+  end
+
+  def change_to_vin_key(vehicle_info)
+    if vehicle_info.has_key?(:vin)
+      vehicle_info[:vin]
+    elsif vehicle_info.has_key?(:vin_1_10)
+      vehicle_info[:vin_1_10]
+    else
+      'Unavailable.'
+    end
+  end
+
+  def change_to_model_key(vehicle_info)
+    if vehicle_info.has_key?(:model)
+      vehicle_info[:model]
+    else
+      'Unavailable.'
+    end
+  end
+
+  def count_vehicle_makes
+    makes = Hash.new(0)
+    @vehicles.each do |vehicle|
+      makes[vehicle.make] += 1
+    end
+    makes
+  end
```

Comment:
Nice job here. Look into the reduce enumerable, see if you can rewrite this using that one! It's fun.